core/trino-main/src/main/java/io/trino/operator/OperatorContext.java

```diff
@@ -231,6 +232,11 @@ public void setLatestMetrics(Metrics metrics)
         this.metrics.set(metrics);
     }
 
+    public void setLatestConnectorMetrics(Metrics metrics)
```

Comment:
Does it make sense for this to live in OperatorContext? It's only relevant to scan operators.

Reply:
Where would it live otherwise? Source operators need to set it for `OperatorsStats` to be propagated.

Reply:
Yes, good point :)